src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -86,6 +88,16 @@ protected function instantiateController($class)
             return $this->container->get($class);
         }
 
-        return parent::instantiateController($class);
+        $e = null;
+
+        try {
+            return parent::instantiateController($class);
+        } catch (FatalThrowableError $e) {
+        } catch (\ArgumentCountError $e) {
+        }
+
+        if (null !== $e && $this->container instanceof Container && in_array($class, $this->container->getRemovedIds())) {
+            throw new \InvalidArgumentException(sprintf('The controller %s seems to be removed from the container, which means that he maybe private, add the tag \'controller.service_arguments\' or made it public.', $class));
```

Comment:
RuntimeException? The argument is valid, the config is not.
Also pass $e as previous exception to the constructor?